llvm/lib/ObjectYAML/MinidumpYAML.cpp

```diff
@@ -373,7 +373,6 @@ void yaml::MappingContextTraits<MemoryDescriptor, yaml::BinaryRef>::mapping(
 void yaml::MappingContextTraits<MemoryDescriptor_64, yaml::BinaryRef>::mapping(
     IO &IO, MemoryDescriptor_64 &Memory, BinaryRef &Content) {
   mapRequiredHex(IO, "Start of Memory Range", Memory.StartOfMemoryRange);
-  mapRequiredHex(IO, "Data Size", Memory.DataSize);
```

Comment:
I actually think we should provide a way to override the DataSize field, as the main purpose of the yaml is to test the parser with possibly invalid input. It doesn't have to be present by default though, which I think you can achieve by using `IO.mapOptional("Data Size", Memory.DataSize, Content.binary_size());` -- and putting it after the "Content" line.
(I'm not using hex, because the RawContentStream is also printed in decimal, but I could be convinced to change both).

Reply:
I gave this a try, but it appears we'll need to do more than set the backup to `Contnet.binary_size()` in the mapping. Content isn't initialized when the mapping is being configured, so `binary_size()` always resolves to 0.

We could do something like in the yaml/minidump emitter if DataSize is 0 we override it, but currently my patch sticks to setting datasize to the content bytesize when it's being emitted.

Reply:
> Content isn't initialized when the mapping is being configured, so binary_size() always resolves to 0.

Are you sure about that? Did you put this line after the line that initializes the `Content` field?

This pattern really should work, as that's the same one used for RawContentStream. And now that I've looked into it more closely, I think it'd be best to do exactly the same thing as RawContentStream does, where a smaller DataSize field is an error, and a larger value just pads the output with zeroes (so if you just want a field of a certain size, and don't care about the contents, you can just put `DataSize: 47` and omit the Content entirely:

```
$ yaml2obj <<EOF | obj2yaml
--- !minidump
Streams:
  - Type:            LinuxAuxv
    Size:            100
    Content:         DEADBEEFBAADF00D
EOF
--- !minidump
Streams:
  - Type:            LinuxAuxv
    Content:         DEADBEEFBAADF00D0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000
...

$ yaml2obj <<EOF
--- !minidump
Streams:
  - Type:            LinuxAuxv
    Size:            7
    Content:         DEADBEEFBAADF00D
EOF                                  
YAML:3:5: error: Stream size must be greater or equal to the content size
  - Type:            LinuxAuxv
    ^
yaml2obj: error: failed to parse YAML input: Invalid argument

$ yaml2obj <<EOF | obj2yaml
--- !minidump
Streams:                      
  - Type:            LinuxAuxv
    Size:            10 
EOF                                  
--- !minidump
Streams:
  - Type:            LinuxAuxv
    Content:         '00000000000000000000'
...

```

Reply:
> Are you sure about that?

No 😆, I trust your knowledge of this much more than my inability to understand. I will try again and use RawContnetStream as a reference this time.